chaincode/traintuple_composite_test.go

```diff
@@ -163,9 +163,14 @@ func TestTraintupleComputePlanCreationComposite(t *testing.T) {
 	args = inpTraintuple.createDefault()
 	resp = mockStub.MockInvoke("42", args)
 	require.EqualValues(t, 400, resp.Status, "should failed for invalid rank")
-	require.Contains(t, resp.Message, "invalid inputs, a new ComputePlan should have a rank 0")
+	require.Contains(t, resp.Message, "Field validation for 'ComputePlanID' failed on the 'required_with' tag")
```

Comment:
Why do this message change ?

Reply:
Before, it was possible to create a new traintuple with Rank `0` and compute plan `null`. In that case, the chaincode would create a new compute plan and assign it to the traintuple.

After this PR, this is not possible because the chaincode is not allowed to issue new keys (all the keys are issued by the backend). So instead, when the backend receives an instruction (by the client) to create a traintuple with rank 0 and compute plan `null`, it explicitly instructs the peer to create a compute plan, **then** creates a traintuple associated with this compute plan (2 ledger invokes instead of 1 previously).

So now there are more invalid scenarios:

- invalid before: compute plan null and rank != 0
- invalid now:  compute plan null and rank not null

The validation now happens at the input validation level using `required_with` (see https://github.com/SubstraFoundation/substra-chaincode/pull/128#discussion_r513776169)